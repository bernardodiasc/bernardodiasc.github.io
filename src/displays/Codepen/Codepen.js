import React, { PureComponent } from 'react'
import './Codepen.css'

class Codepen extends PureComponent {
  static defaultProps = {
    hash: '',
    height: '',
    title: '',
  }

  render() {
    const { hash, height, title } = this.props
    return (
      <iframe
        className="Codepen"
        height={height}
        scrolling="no"
        title={title}
        src={`//codepen.io/bernardodiasc/embed/${hash}/?height=${height}&theme-id=0&default-tab=result&embed-version=2`}
        frameBorder="no"
        allowTransparency="true"
        allowFullScreen="true"
      >
        See the Pen <a href={`https://codepen.io/bernardodiasc/pen/${hash}/`}>{title}</a> by Bernardo Dias (<a href="https://codepen.io/bernardodiasc">@bernardodiasc</a>) on <a href="https://codepen.io">CodePen</a>.
      </iframe>
    )
  }
}

export default Codepen
