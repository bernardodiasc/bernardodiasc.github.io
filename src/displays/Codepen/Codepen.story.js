import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import Codepen from './Codepen'
import CodepenReadme from './README.md'
storiesOf('Displays/Codepen', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{CodepenReadme}</Docs>)
  .add('Example', () => <Codepen />)
  .add('Minimal render', () => <Codepen />)
