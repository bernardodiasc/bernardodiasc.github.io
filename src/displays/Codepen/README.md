# Codepen

## How to use

```
import Codepen from 'displays/Codepen'
```

```
<Codepen />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
