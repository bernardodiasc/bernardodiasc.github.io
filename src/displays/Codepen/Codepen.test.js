import React from 'react'
import { shallow } from 'enzyme'

import Codepen from './Codepen.js'

it('renders without crashing', () => {
  const component = shallow(<Codepen />)
  expect(component).toMatchSnapshot()
})
