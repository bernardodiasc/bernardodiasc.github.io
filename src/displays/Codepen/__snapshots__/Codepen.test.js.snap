// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<iframe
  allowFullScreen="true"
  allowTransparency="true"
  className="Codepen"
  frameBorder="no"
  height=""
  scrolling="no"
  src="//codepen.io/bernardodiasc/embed//?height=&theme-id=0&default-tab=result&embed-version=2"
  title=""
>
  See the Pen 
  <a
    href="https://codepen.io/bernardodiasc/pen//"
  />
   by Bernardo Dias (
  <a
    href="https://codepen.io/bernardodiasc"
  >
    @bernardodiasc
  </a>
  ) on 
  <a
    href="https://codepen.io"
  >
    CodePen
  </a>
  .
</iframe>
`;
